provider/testdata/src/lib.rs

```diff
@@ -128,73 +156,76 @@ pub mod versions {
 #[cfg(feature = "std")]
 pub mod paths;
 
-#[cfg(feature = "static")]
-use {
-    icu_provider_adapters::fallback::LocaleFallbackProvider, icu_provider_blob::StaticDataProvider,
-};
+use icu_provider::prelude::*;
+use icu_provider_adapters::fallback::LocaleFallbackProvider;
 
-#[cfg(feature = "fs")]
-use icu_provider_fs::FsDataProvider;
-
-/// Get a data, loading from the test data JSON directory.
-///
-/// You can optionally specify your own test data with the
-/// `ICU4X_TESTDATA_DIR` environment variable.
+/// A data provider that is compatible with all ICU `_unstable` constructors.
 ///
-/// # Panics
+/// The return type of this method is not considered stable, mirroring the unstable trait
+/// bounds of the constructors. For matching versions of `icu` and `icu_testdata`, however,
+/// these are guaranteed to match.
+pub fn unstable() -> LocaleFallbackProvider<UnstableDataProvider> {
+    // The statically compiled data file is valid.
+    #[allow(clippy::unwrap_used)]
+    LocaleFallbackProvider::try_new_unstable(unstable_no_fallback()).unwrap()
+}
+
+/// A data provider that is compatible with all ICU `_unstable` constructors.
 ///
-/// Panics if unable to load the data.
-// The function is documented to allow panics.
-#[allow(clippy::panic)]
-#[cfg(feature = "fs")]
-pub fn get_json_provider() -> FsDataProvider {
```

Comment:
Issue: There's nothing to replace `FsDataProvider` and `get_json_provider`.

We have the data, and we should expose a canonical provider for it. Please add it back.

Perhaps `icu_testdata::json()` and `icu_testdata::json_no_fallback()`

Reply:
I think it's fine to have JSON only as a human-readable postcard version for code reviews, and don't see the need to expose it just because we have it. It's the exact same data as postcard, and nobody is using it (why would they, it's inefficient and might require extra features).